beamer/chain.py

```diff
@@ -115,11 +115,11 @@ def __init__(self, context: Context):
     @property
     def _synced(self) -> bool:
         with self._lock:
-            return self._num_syncs_done == 2
+            return self._num_syncs_done == 3
 
     def mark_sync_done(self) -> None:
         with self._lock:
-            assert self._num_syncs_done < 2
+            assert self._num_syncs_done < 3
```

Comment:
I don't know how but is it possible to not have this hard coded value here? Subject to errors in the future if we add contract monitors. 

Maybe one solution could be capturing all contract monitors in a dict. 

```py
self._contract_monitors = {
    "request_manager": ContractMonitor(...),
    "fill_manager": ....
    ....
    }
```

and then at the respective comparisons, you could use `len(self._contract_monitors)`. Just an idea...

Reply:
ANother alternative is that we make each event monitor able to query multiple contracts. That would also help to keep the number of RPC requests down

Reply:
> ANother alternative is that we make each event monitor able to query multiple contracts. That would also help to keep the number of RPC requests down

Only if all the contracts are on the same chain, right?

Reply:
Of course, we need at least one for the source and one for the target chain

Reply:
Hm, that sounds like a nice optimisation. I like it. Even more so if one considers the plan to support multiple chain pairs with a single agent. In that case, tracking all event monitors would become a bit messy, but with your approach we would only ever have 1 event monitor per supported chain, which makes scaling as good as it can possibly get. That's a definite like. :+1:

Reply:
Done.